### ISSUE TITLE
copy for slack doesn't use rich text

### ISSUE THREAD

**Issue by Geczy** [abachman/vscode-copyfordocumentation#3]
any idea what's happening here?

![CleanShot 2024-07-17 at 08 53 33](https://github.com/user-attachments/assets/9ddf016c-5165-4b39-9144-7b05d0425218)

**Comment by abachman** [comment]
ooo, this could be a difference in our Slack settings.

I'm only using the plugin heavily for one account at the moment, and I've enabled `[x] Format messages with markup` for it. That's under the "Advanced" Preferences menu, "Input options" section. This is on the desktop Slack.app on macos. 

![Screenshot 2024-07-22 at 9 02 11 AM](https://github.com/user-attachments/assets/b84f3029-a716-4bf6-a2b6-adf6a996fa4c)

mostly related: A "Copy as Rich Text" option would be interesting. I've used [this page / app for editing Google Slides presentations](https://romannurik.github.io/SlidesCodeHighlighter/), I suspect it wouldn't be too awful to implement the same for VS Code.

**Comment by Geczy** [comment]
yep it works with that option checked. i wish there was a quick way to toggle it off and on because i use it as well for some other generated messages, but turn it back off when i'm done and want to just type messages myself with keyboard rich text shortcuts

**Comment by Geczy** [comment]
i just realized that if you copy any text in a `<pre>`, like on your browser, and paste it to slack, it will paste as a code block

so i'm wondering if this vscode extension can copy with rich text instead of markdown? 

see vid:
![CleanShot 2024-07-24 at 07 44 43](https://github.com/user-attachments/assets/2c7d009a-7242-4540-805b-1916e3d9632b)

**Comment by Geczy** [comment]
this works to copy both rich text and plain text to keyboard, letting mac decide which one to paste depending on source:

```
plainText="Your plain text here"

htmlContent="<b>bolded text</b>" # Example HTML content

# Convert HTML content to hex for the clipboard
htmlHex=$(echo "$htmlContent" | hexdump -ve '1/1 "%.2x"')

# Use AppleScript to set both the plain text and the HTML content of the clipboard
osascript -e "set the clipboard to {text:\"${plainText}\", «class HTML»:«data HTML${htmlHex}»}"
```
